### PR TITLE
Change move item todo API

### PR DESCRIPTION
**Pull request by bramkragten**
## Proposed change

For https://github.com/home-assistant/core/pull/102795

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/lovelace/cards/hui-todo-list-card.ts [hunk 1]
```diff
@@ -439,10 +438,36 @@ export class HuiTodoListCard
...
+    this._items!.splice(itemIndex, 1);
+    if (newIndex === 0) {
+      this._items!.unshift(item);
+    } else {
+      const prevIndex = this._items!.findIndex(
+        (itm) => itm.uid === prevItem!.uid
+      );
+      this._items!.splice(prevIndex + 1, 0, item);
+    }
+    this._items = [...this._items!];
```
Comment (allenporter) [inline src/panels/lovelace/cards/hui-todo-list-card.ts:463]: Is this just making another copy? Seems good just unclear to me as a novice why this is needed.

**Review by allenporter — APPROVED** [review]

File: src/panels/lovelace/cards/hui-todo-list-card.ts (hunk 1 above)
Comment (bramkragten) [reply src/panels/lovelace/cards/hui-todo-list-card.ts:463]: It is to force lit to update, otherwise it thinks `_items` is not changed.

Comment (piitaya) [reply src/panels/lovelace/cards/hui-todo-list-card.ts:463]: You can also do this right?
```ts
this.requestUpdate("_items");
```

**Review by piitaya — DISMISSED** [review]

File: src/panels/lovelace/cards/hui-todo-list-card.ts (hunk 1 above)
Comment (bramkragten) [reply src/panels/lovelace/cards/hui-todo-list-card.ts:463]: > You can also do this right?
> 
> ```ts
> this.requestUpdate("_items");
> ```

Not sure if that works with repeat

Comment (bramkragten) [reply src/panels/lovelace/cards/hui-todo-list-card.ts:463]: Nope, it doesn't

**Review by piitaya — APPROVED** [review]

**Pull request merged**